src/components/DistanceRequest.js

```diff
@@ -146,28 +144,21 @@ function DistanceRequest({transactionID, report, transaction, mapboxAccessToken,
         [waypoints, lastWaypointIndex],
     );
 
-    // Show up to the max number of waypoints plus 1/2 of one to hint at scrolling
-    const halfMenuItemHeight = Math.floor(variables.optionRowHeight / 2);
-    const scrollContainerMaxHeight = variables.optionRowHeight * MAX_WAYPOINTS_TO_DISPLAY + halfMenuItemHeight;
-
     useEffect(() => {
         MapboxToken.init();
         return MapboxToken.stop;
     }, []);
 
     useEffect(() => {
-        if (!transactionID || !_.isEmpty(waypoints)) {
+        const transactionWaypoints = lodashGet(transaction, 'comment.waypoints', {});
+        if (!transactionID || !_.isEmpty(transactionWaypoints)) {
```

Comment:
when `transactionID` is defined (i.e. when the transaction ID is stored inside iou), are we 100% sure that the `waypoints` are also defined?

Reply:
Hey, we have investigated it and found some ways to improve it:

**Before:**
Previously, we unnecessarily loaded the whole `transaction` collection (it was because `transactionID` was undefined). Moreover, because of that we also called the `createInitialWaypoints` twice.
![Screenshot 2023-10-04 at 11 11 22](https://github.com/Expensify/App/assets/104823336/a27f9305-bd40-4e6b-805b-61a16db41980)

**After:**
I used a similar approach to [this](https://github.com/Expensify/App/blob/0b3b3ec7f569a4b3c401c3a322dcc2b6debd3862/src/components/ReportActionItem/MoneyRequestPreview.js#L377) and [this](https://github.com/Expensify/App/blob/0b3b3ec7f569a4b3c401c3a322dcc2b6debd3862/src/components/MoneyRequestHeader.js#L148) one. I also added a check if the `transaction` (with the waypoints) is 100% sure loaded. [Related commit.](https://github.com/Expensify/App/pull/26307/commits/970a3b034a598ccbaf33b9da8f02ca67468fb0c3)

Results:
![Screenshot 2023-10-04 at 11 31 02](https://github.com/Expensify/App/assets/104823336/9ce373a8-91b6-4652-abcf-37e7b8c4f232)

Now:
- we do not load the whole `transaction` collection from Onyx,
- we have only one `createInitialWaypoints` call.

Reply:
ah nice! I didn't know we loaded the whole `transactions` collection while the `transactionID` was undefined.

nice catch!

Reply:
It might be worth checking in other components as well to see if this issue doesn't occur.